nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/ProtectedNiFiPropertiesGroovyTest.groovy

```diff
@@ -77,7 +83,7 @@ class ProtectedNiFiPropertiesGroovyTest extends GroovyTestCase {
         }
     }
 
-    private static ProtectedNiFiProperties loadFromFile(String propertiesFilePath) {
+    private static ProtectedNiFiProperties loadFromFile(String propertiesFilePath, String defaultKeyHex) {
```

Comment:
Change `keyHex` to provider instance as with application code.

Reply:
I did this in two steps:

1.  Replaced all of the string parameters to the `ProtectedNiFiProperties` ctors and factory methods with `SensitivePropertyProvider` parameters.  Adjusted all call sites with new parameter instances.

2.  Added new ctors and factory method signatures to support passing strings as keys.  These methods construct the `StandardSensitivePropertyProvider` instance just like the client code was doing, no keys or references to keys are saved.